### DISCUSSION TITLE
rc on main branch and pr on prs

### DISCUSSION THREAD

**Discussion by itewk** [callowayproject/bump-my-version#250] [Q&A]
First, this tool is awesome and I hope to use it to replace bash script i wrote a long time ago.

What I am trying to do is having a worklow that looks like:

main
|- v0.1.0
|- v0.2.0-rc0
|- v0.2.0-rc1

create PR #32 from branch feature/foo

feature/foo (pr #42)
|- v0.2.0-pr42.0
|- v0.2.0-pr42.1
|- v0.2.0-pr42.2

merge pr #42
main
|- v0.1.0
|- v0.2.0-rc0
|- v0.2.0-rc1
|- v0.2.0-rc2

release main
main
|- v0.1.0
|- v0.2.0-rc0
|- v0.2.0-rc1
|- v0.2.0-rc2
|- v0.2.0

I can't figure out the right way to do the parse and serialize to get this behavor. I can get the main branch rc behavor using the example from the docs. I have tried using a modified version of https://github.com/callowayproject/bump-my-version/blob/master/tests/fixtures/pep440.toml to do the pr tagging but with no luck.

Anyone do anythuing like this?

**Comment by itewk** [comment]
@coordt i saw on another post that you don't get notifications for new posts. My apologies if I am out of step tagging you here and this is not meant as a "please answer. me right now" more just a "hey, fyi, this is here when/if you have an opportunity to take a look". 

And thanks for this great tool.

**Comment by coordt** [comment]
First off, thanks for the support. I appreciate it!

I'm unsure if I can give you a specific solution because I don't truly understand your desired workflow. But I will give you some guidance that might help.

Two things to remember: the actual version always includes all the segments, and a segment typically depends on the previous segment. 

How does the `pr` segment depend on the `rc` segment, and how do they both depend on the `patch` segment?

For example, does version `0.2.0-pr42.1` break down into segments:

- **major:** 0
- **minor:** 1
- **patch:** 0
- **pre_l:** <empty string>
- **pre_n:** 0
- **dev_l:** pr
- **dev_n:** 1

After looking at this, you might be able to get what you want by splitting the `dev` segment in the [PEP440 pattern](https://github.com/callowayproject/bump-my-version/blob/a5fc5c0e595530650059dd6ab821927933f0ef58/bumpversion/versioning/conventions.py#L1-L1) into `dev_l` and `dev_n` segments similar to the `pre_l` and `pre_n` segments.

I will warn you that you may regret a complex versioning convention. It means you will have to put more effort into managing it.

Let me know if I helped at all.
